### ISSUE TITLE
Way to clear cache

### ISSUE THREAD

**Issue by djsegal** [JuliaCollections/Memoize.jl#14]
For example,

+ let's say you need to do a few batches of jobs
+ the functions need to be cleared out after each batch

-------

are you supposed to use a dictionary as the key store and clear it out each time?

**Comment by djsegal** [comment]
way around it:

```
function dememoize()

  mod = current_module()

  all_caches = filter( x ->
    startswith(string(x), "##") &&
    endswith(string(x),"_memoized_cache"),
    names(mod, true)
  )

  for cur_cache_symbol in all_caches
    empty!(getfield(mod, cur_cache_symbol))
  end

end
```

**Comment by colinfang** [comment]
@djsegal  FYI, I have an implementation of [Memorize.jl](https://github.com/colinfang/Memoize.jl) that does clear cache.

**Comment by djsegal** [comment]
Are you planning on renaming it and submitting it to `METADATA.jl`?

**Comment by PharmCat** [comment]
Is a way to clear cache now? 
I think Momoize leads to a memory leak  without clear cache function.

**Comment by cstjean** [comment]
Fixed by #60
